tests/app/clients/test_aws_sns.py

```diff
@@ -30,3 +30,24 @@ def test_send_sms_returns_raises_error_if_there_is_no_valid_number_is_found(noti
         aws_sns_client.send_sms(to, content, reference)
 
     assert 'No valid numbers found for SMS delivery' in str(excinfo.value)
+
+
+def test_send_sms_with_long_code_successful_returns_aws_sns_response(notify_api, mocker):
```

Comment:
Main new test for the SNS provider, making sure that if a sender is specified, the `_long_codes_client` is used and the origination number is specified